launcher/minecraft/mod/ResourcePackFolderModel.h

```diff
@@ -17,7 +17,7 @@ class ResourcePackFolderModel : public ResourceFolderModel
         NUM_COLUMNS
     };
 
-    explicit ResourcePackFolderModel(const QString &dir, BaseInstance* instance);
+    explicit ResourcePackFolderModel(const QString& dir, const BaseInstance* instance);
```

Comment:
the non-const nest of the instance pointer was intentional because the model can modify the instance's settings
was there a way around this?

Reply:
The model doesn't change the `BaseInstance`.
For example `ResourcePackFolderModel` only sends the instance down to `ResourceFolderModel` and doesn't use it. And `ResourceFolderModel` is using it in a const function (https://github.com/PrismLauncher/PrismLauncher/blob/develop/launcher/minecraft/mod/ResourceFolderModel.cpp#L559-L561). 
This is how all the Models that derive from `ResourceFolderModel` use the BaseInstance.
The only exception is `WorldList` which also uses the instance in the same function as  `ResourceFolderModel`(same signature, same implementation).

But if needed, I can remove the const and have it propagated in the needed classes as a non-const value.

Reply:
oh, the function I was thinking of hasn;t been merged yet. it's in #1058

https://github.com/PrismLauncher/PrismLauncher/pull/1058/files#diff-cb589c4e3cd01168d0845fb44ff24dc054c58b1f7a6c7637a653a5a6d527e5cbR516

Reply:
Fair enough, reverted the const change and removed the const from ModModel